tools/generators/swift_debug_settings/src/Generator/WriteSwiftDebugSettings.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+import PBXProj
+
+extension Generator {
+    struct WriteSwiftDebugSettings {
+        private let write: Write
+
+        private let callable: Callable
+
+        /// - Parameters:
+        ///   - callable: The function that will be called in
+        ///     `callAsFunction()`.
+        init(
+            write: Write,
+            callable: @escaping Callable = Self.defaultCallable
+        ) {
+            self.write = write
+
+            self.callable = callable
+        }
+
+        /// Writes the Swift debug settings to disk.
+        func callAsFunction(
+            _ keyedSwiftDebugSettings:
+                [(key: String, settings: TargetSwiftDebugSettings)],
+            to url: URL
+        ) throws {
+            try callable(
+                /*keyedSwiftDebugSettings:*/ keyedSwiftDebugSettings,
+                /*url:*/ url,
+                /*write:*/ write
+            )
+        }
+    }
+}
+
+// MARK: - WriteSwiftDebugSettings.Callable
+
+extension Generator.WriteSwiftDebugSettings {
+    typealias Callable = (
+        _ keyedSwiftDebugSettings:
+            [(key: String, settings: TargetSwiftDebugSettings)],
+        _ url: URL,
+        _ write: Write
+    ) throws -> Void
+
+    static func defaultCallable(
+        _ keyedSwiftDebugSettings:
+            [(key: String, settings: TargetSwiftDebugSettings)],
+        to url: URL,
+        write: Write
+    ) throws {
+        let content = #"""
+#!/usr/bin/python3
+
+"""An lldb module that registers a stop hook to set swift settings."""
+
+import lldb
+import re
+
+# Order matters, it needs to be from the most nested to the least
+_BUNDLE_EXTENSIONS = [
+    ".framework",
+    ".xctest",
+    ".appex",
+    ".bundle",
+    ".app",
+]
+
+_TRIPLE_MATCH = re.compile(r"([^-]+-[^-]+)(-\D+)[^-]*(-.*)?")
+
+_SETTINGS = {
+\#(keyedSwiftDebugSettings.map(settingsString).joined())\#
```

Comment:
Just to understand: why not generate json files with the settings and pass them as an argument to one Python script instead of generating Python scripts?